### PR TITLE
Use i18n update script when generating JS bundle

### PR DESCRIPTION
**Pull request by fluiddot**
**Depends on the following PRs:**
- Gutenberg PR: https://github.com/WordPress/gutenberg/pull/37787.
- Docker GB-mobile image PR: https://github.com/wordpress-mobile/docker-gb-mobile-image/pull/4

This PR follows up on https://github.com/wordpress-mobile/gutenberg-mobile/pull/4366 and attaches the i18n update script to the command used for generating the JS bundle. 

Additionally, it removes the folder `src/i18n-translations`, which contained the static translations generated on https://github.com/wordpress-mobile/gutenberg-mobile/pull/4359, and use folder `src/i18n-cache` with the translations generated on the fly when producing the JS bundle with i18n update script.

**UPDATE:** A new script for checking the i18n cache upon dependencies install has been introduced ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/40fa7390b95a6d6db2b148ccadff7abc6e62ad9e/bin/i18n-check-cache.sh)), the purpose of this script is to assure that translation files are always present for all used plugins. Otherwise, the JS bundle generation, as well as Metro server, would throw an exception as these files are referenced upon editor initialization ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/68d7091ea4e6181f511d281c88f0fa25ff3077bc/src/index.js#L9-L10)).

**NOTE:** As a side note, I noticed that the "Bundle Android" Buildkite job ([reference](https://buildkite.com/wordpress-mobile/gutenberg-mobile/builds/1866#c898b8db-2279-46ed-b63d-83b212c4ebd8)) has increased the duration by extra 10 minutes due to the i18n update command. 

### Verify that bundle command executes i18n update command
1. Run command `npm run bundle`
2. Observe that the `i18n:update` command is executed by checking the log output.

**NOTE:** It's expected to see modifications on files under the `bundle` folder. Once the PR is ready to merge, we could update the bundle and push these changes.

### Verify that "Bundle Android" Buildkite job executes i18n update command
1. Check latest Buildkite job ([reference](https://buildkite.com/wordpress-mobile/gutenberg-mobile/builds/1879#9dde74b1-9a36-461b-9091-1afb46888d9e)).
2. Observe that the `i18n:update` command is executed by checking the log output.

### Verify that dependency installation executes i18n check cache command
1. Run command `npm ci`.
2. Observe that the `i18n:check-cache` command is executed by checking the log output.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: bundle/android/strings.xml [hunk 1]
```diff
@@ -169,6 +169,7 @@
     <string name="gutenberg_native_file_block_settings" tools:ignore="UnusedResources">File block settings</string>
     <string name="gutenberg_native_file_name" tools:ignore="UnusedResources">File name</string>
     <string name="gutenberg_native_font_size" tools:ignore="UnusedResources">Font Size</string>
+    <string name="gutenberg_native_four" tools:ignore="UnusedResources">Four</string>
```
Comment (fluiddot) [inline bundle/android/strings.xml:171]: I spotted that Layout Grid uses strings from the default domain instead of using `layout-grid`, not sure whether this is expected so I posted a comment for further info ([reference](https://github.com/Automattic/block-experiments/pull/187/files#r776415460)). In any case, these strings are not included in any GlotPress project ([Gutenberg](https://translate.wordpress.org/projects/wp-plugins/gutenberg/stable/es/default/?filters%5Bterm%5D=Four&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc) / [Layout Grid](https://translate.wordpress.org/projects/wp-plugins/layout-grid/stable/es/default/?filters%5Bterm%5D=Four&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc)) so makes sense to translate them as native ones.

File: .gitignore [hunk 2]
```diff
@@ -120,3 +120,6 @@ bin/wp-cli.phar
 
 ios/Pods
 ios/vendor
+
+# i18n update
+i18n-test/
```
Comment (fluiddot) [inline .gitignore:125]: This is a test folder we could use when testing the i18n update command.

File: bin/i18n-update.sh [hunk 3]
```diff
@@ -80,25 +96,41 @@ fi
...
 
+# Check if the process should be skipped due to the i18n cache presence
+if [[ -n ${SKIP_IF_CACHE:-} ]]; then
+  if [[ $(arrayLength "${NOT_FOUND_PLUGIN_I18N_CACHE[@]+"${NOT_FOUND_PLUGIN_I18N_CACHE[@]}"}") -eq 0 ]]; then
+    echo -e "Skipping the update due to SKIP_IF_CACHE enabled."
+    exit 0
+  else
+    echo -e "\n\033[1mForcing the update due to some cache folders not found.\033[0m"
+  fi
+fi
```
Comment (fluiddot) [inline bin/i18n-update.sh]: `SKIP_IF_CACHE` option basically checks if we already have a cache folder for each plugin, if not, it will force updating the localizations.

File: bin/i18n-update.sh [hunk 4]
```diff
@@ -27,19 +27,19 @@ while test $# -gt 0; do
   case "$1" in
     -h|--help)
       echo "options:"
-      echo "-h, --help    show brief help"
-      echo "-p, --path    local path for generating files (by default a temp folder will be used)"
-      echo "-d, --debug   print extra info for debugging"
```
Comment (fluiddot) [inline bin/i18n-update.sh:32]: I removed the `debug` option as it wasn't being used.

File: bin/i18n-update.sh [hunk 5]
```diff
@@ -27,19 +27,19 @@ while test $# -gt 0; do
...
+      echo "-h, --help              show brief help"
+      echo "-p, --path              local path for generating files (by default a temp folder will be used)"
+      echo "-s, --skip-if-cache     skip update if cache is present (i.e. \"src/i18n-cache\" folder)"
```
Comment (fluiddot) [inline bin/i18n-update.sh]: `skip-if-cache` option will allow us to only update the i18n localizations in case we're missing any of the cache folders of the plugins, this way we can attach it to dependency installation.

File: bin/i18n-update.sh [hunk 6]
```diff
@@ -80,25 +102,45 @@ fi
...
+    echo -e "-- Skipping the update due to SKIP_IF_CACHE enabled --"
+
+    # Even though localizations are not updated, we need to guarantee that i18n cache within react-native-editor package is updated
+    update_gutenberg_i18n_cache "$TRANSLATIONS_OUTPUT_PATH"
+
+    exit 0
+  else
+    echo -e "\n\033[1mForcing the update due to some cache folders not found.\033[0m"
+  fi
+fi
```
Comment (fluiddot) [inline bin/i18n-update.sh]: We need to assure that we have translations in the cache folders (`src/i18n-cache`) in order to determine if the script should continue or not.

File: package.json [hunk 7]
```diff
@@ -53,7 +53,7 @@
 		"wd": "^1.11.1"
 	},
 	"scripts": {
-		"postinstall": "patch-package && npm ci --prefix gutenberg && cd jetpack/projects/plugins/jetpack && npx pnpm install",
+		"postinstall": "patch-package && npm run i18n:update:skip-if-cache && npm ci --prefix gutenberg && cd jetpack/projects/plugins/jetpack && npx pnpm install",
```
Comment (fluiddot) [inline package.json]: Translations are required for running the editor, and since they are not versioned, it's important that we guarantee their presence when installing the dependencies. This behavior follows the same approach we do on Gutenberg ([reference](https://github.com/WordPress/gutenberg/blob/fe9b08a2fd5dacb5948c3c5fc1af875f84e2142e/packages/react-native-editor/package.json#L95)).

File: package.json [hunk 8]
```diff
@@ -62,37 +62,15 @@
 		"rn-bundle": "react-native bundle",
 		"postrn-bundle": "patch-package --reverse --patch-dir gutenberg/packages/react-native-editor/metro-patch",
 		"i18n:update": "./bin/i18n-update.sh jetpack ./jetpack/projects/plugins/jetpack/extensions layout-grid ./block-experiments/blocks/layout-grid",
-		"i18n:update:debug": "./bin/i18n-update.sh --debug jetpack ./jetpack/projects/plugins/jetpack/extensions layout-grid ./block-experiments/blocks/layout-grid",
-		"i18n:gutenberg": "./bin/i18n-download.sh gutenberg",
-		"i18n:gutenberg:force": "cross-env REFRESH_I18N=1 ./bin/i18n-download.sh gutenberg",
-		"bundle": "npm run clean; npm run bundle:js && npm run genstrings",
-		"prebundle:js": "npm run i18n:gutenberg:force",
+		"i18n:update:skip-if-cache": "./bin/i18n-update.sh --skip-if-cache jetpack ./jetpack/projects/plugins/jetpack/extensions layout-grid ./block-experiments/blocks/layout-grid",
+		"i18n:update:test": "./bin/i18n-update.sh --path i18n-test jetpack ./jetpack/projects/plugins/jetpack/extensions layout-grid ./block-experiments/blocks/layout-grid",
```
Comment (fluiddot) [inline package.json:66]: This is an extra command I added in case we want to check the intermediate files produced when updating localizations.

File: package.json [hunk 9]
```diff
@@ -53,46 +53,24 @@
...
-		"makepot:layout-grid": "npm run wp -- i18n make-pot ./block-experiments/blocks/layout-grid --exclude=test/*,e2e-tests/*,build/*,build-module/*,build-style/*,*.native.js,*.ios.js,*.android.js,bundle/* --domain=layout-grid layout-grid.pot",
-		"pregenstrings": "npm run makepot:gutenberg && npm run makepot:jetpack && npm run makepot:layout-grid",
-		"genstrings": "npm run genstrings:android && npm run genstrings:ios",
-		"genstrings:android": "npm run makepot:android:gutenberg && npm run makepot:android:gutenberg-mobile && npm run makepot:android:jetpack && npm run makepot:android:layout-grid && ./bin/po2android.js bundle/android/strings.xml gutenberg-android.pot gutenberg-mobile-android.pot jetpack-android.pot layout-grid-android.pot",
-		"genstrings:ios": "npm run makepot:ios:gutenberg && npm run makepot:ios:gutenberg-mobile && npm run makepot:ios:jetpack && npm run makepot:ios:layout-grid && ./bin/po2swift.js bundle/ios/GutenbergNativeTranslations.swift gutenberg-ios.pot gutenberg-mobile-ios.pot jetpack-ios.pot layout-grid-ios.pot",
```
Comment (fluiddot) [inline package.json:95]: These commands are no longer required as this logic is covered in the `generate-pot-files` script ([reference](https://github.com/WordPress/gutenberg/blob/trunk/packages/react-native-editor/bin/generate-pot-files.sh)).

File: .buildkite/pipeline.yml [hunk 10]
```diff
@@ -14,9 +14,12 @@ steps:
     key: "bundle-android"
     plugins:
       - docker#v3.8.0:
-          image: "public.ecr.aws/automattic/gb-mobile-image:latest"
+          image: "public.ecr.aws/automattic/gb-mobile-image:6804aad7632be7db25870cdf283ed310f0c8380e"
```
Comment (fluiddot) [inline .buildkite/pipeline.yml]: Once https://github.com/wordpress-mobile/docker-gb-mobile-image/pull/4 is merged and a new Docker image version is released, we should revert this change.

Comment (fluiddot) [reply .buildkite/pipeline.yml]: The PR was merged so the version has been changed to `latest` in https://github.com/wordpress-mobile/gutenberg-mobile/pull/4411/commits/40fa7390b95a6d6db2b148ccadff7abc6e62ad9e.

File: bin/i18n-update.sh (hunk 6 above)
Comment (fluiddot) [reply bin/i18n-update.sh]: This was addressed by adding the `i18n-check-cache` script ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/40fa7390b95a6d6db2b148ccadff7abc6e62ad9e/bin/i18n-check-cache.sh)).

File: bin/i18n-update.sh (hunk 5 above)
Comment (fluiddot) [reply bin/i18n-update.sh]: This option is no longer required as the logic related to it has been introduced in [this script](https://github.com/wordpress-mobile/gutenberg-mobile/blob/40fa7390b95a6d6db2b148ccadff7abc6e62ad9e/bin/i18n-check-cache.sh).

**Review by jhnstn — APPROVED** [review]
Changes look good, Test plan ran as expected
